lightning/src/ln/channelmanager.rs

```diff
@@ -4241,6 +4251,37 @@ where
 		Ok(())
 	}
 
+	/// Gets the number of peers which match the given filter which have unfunded inbound channels.
+	///
+	/// The filter is called for each node and provided with the number of unfunded channels the
+	/// node has.
+	///
+	/// If the peer in question has funded channels, however, zero will be returned.
+	fn peers_with_unfunded_channels<Filter>(&self, mut filter: Filter) -> usize
```

Comment:
I think this function is a little too complex with the filter.  I found it difficult to follow the code, so I'm afraid bugs may hide in here (if not now, in the future).

Reply:
I don't think `peers_with_unfunded_channels` is better or worst than few more critical code paths we have in `onchaintx.rs` / `package.rs` and other locations. Code quality sounds good enough to me.

Reply:
I think we could improve readability with a couple simple changes.  We could create 2 new helper functions `has_funded_channel(node_id)` and `unfunded_channel_count(node_id)` and use those instead of extracting that information from the filter functions.  This would then allow us to simplify the filters to one-line functions that only take a `node` parameter.